pkg/azuredevops/pipelines/pipelines.go

```diff
@@ -110,6 +133,278 @@ func GetRunLogs(ctx context.Context, buildClient BuildClient, httpClient HTTPCli
+//
+// Parameters:
+// timeline  - A pointer to a build.Timeline struct containing a slice of build records.
+// testName  - The name of the test to look for within the build records.
+// testResult - The expected result of the test (e.g., "Succeeded", "Skipped").
+// testState  - The expected state of the test (e.g., "Completed", "Pending").
+//
+// Returns a boolean and an error. The boolean is true if a record matching all the specified criteria (test name, result, and state)
+// is found in the timeline. If no matching record is found, the function returns false and an error indicating the absence of a
+// record that meets the specified conditions.
+//
+// This function is useful for verifying specific outcomes in a series of build tests, particularly for continuous integration and
+// deployment scenarios where test results need to be programmatically verified.
+func CheckBuildRecords(timeline *build.Timeline, testName, testResult, testState string) (bool, error) {
+	for _, record := range *timeline.Records {
+		if record.Name != nil && *record.Name == testName {
+			if record.Result != nil && string(*record.Result) == testResult && record.State != nil && string(*record.State) == testState {
+				return true, nil // Found a record matching all criteria
+			}
+		}
+	}
+
+	return false, fmt.Errorf("no record found matching the criteria")
+}
+
+// RunBuildTest executes a build test within a given context. It uses the specified build client
+// to run tests on a project and pipeline, based on the provided pipeline and build IDs.
+// It evaluates the test condition (presence or absence of a specific command) in the build logs.
+//
+// Parameters:
+// ctx           - The context to control the execution and cancellation of the test.
+// buildClient   - The client interface to interact with the build system.
+// projectName   - The name of the project in which the test is being run.
+// pipelineName  - The name of the pipeline within the project.
+// pipelineID    - The identifier of the pipeline.
+// buildID       - A pointer to an integer storing the build identifier.
+// test          - The build test to be executed, which includes test conditions and expectations.
+//
+// Returns true if the test passes, which includes successful execution and meeting of the test conditions.
+// If the test fails due to an error in execution or if the test conditions are not met, the function
+// logs a fatal error with the test description and the reason for the failure.
+func RunBuildTest(ctx context.Context, buildClient BuildClient, projectName, pipelineName string, pipelineID int, buildID *int, test BuildTest) bool {
+	var pass bool
+	var err error
+
+	if test.ExpectAbsent {
+		pass, err = CheckSpecificBuildForMissingCommand(ctx, buildClient, buildID, projectName, pipelineName, test.LogMessage, pipelineID)
+	} else {
+		pass, err = CheckSpecificBuildForCommand(ctx, buildClient, projectName, pipelineName, test.LogMessage, pipelineID, buildID)
+	}
```

Comment:
The list of arguments should have the same order in case both function will stay longer.

Reply:
Not needed anymore, as the function is merged into one.

CheckBuildLogForMessage